src/mame/layout/segaai.lay

```diff
@@ -0,0 +1,52 @@
+			<element inputtag="TP.~rownum~" inputmask="0x000004" ref="touchpad"><bounds x="2" width="1" y="~rownum~" height="1" /></element>
+			<element inputtag="TP.~rownum~" inputmask="0x000008" ref="touchpad"><bounds x="3" width="1" y="~rownum~" height="1" /></element>
+			<element inputtag="TP.~rownum~" inputmask="0x000010" ref="touchpad"><bounds x="4" width="1" y="~rownum~" height="1" /></element>
+			<element inputtag="TP.~rownum~" inputmask="0x000020" ref="touchpad"><bounds x="5" width="1" y="~rownum~" height="1" /></element>
+			<element inputtag="TP.~rownum~" inputmask="0x000040" ref="touchpad"><bounds x="6" width="1" y="~rownum~" height="1" /></element>
+			<element inputtag="TP.~rownum~" inputmask="0x000080" ref="touchpad"><bounds x="7" width="1" y="~rownum~" height="1" /></element>
+			<element inputtag="TP.~rownum~" inputmask="0x000100" ref="touchpad"><bounds x="8" width="1" y="~rownum~" height="1" /></element>
+			<element inputtag="TP.~rownum~" inputmask="0x000200" ref="touchpad"><bounds x="9" width="1" y="~rownum~" height="1" /></element>
+			<element inputtag="TP.~rownum~" inputmask="0x000400" ref="touchpad"><bounds x="10" width="1" y="~rownum~" height="1" /></element>
+			<element inputtag="TP.~rownum~" inputmask="0x000800" ref="touchpad"><bounds x="11" width="1" y="~rownum~" height="1" /></element>
+			<element inputtag="TP.~rownum~" inputmask="0x001000" ref="touchpad"><bounds x="12" width="1" y="~rownum~" height="1" /></element>
+			<element inputtag="TP.~rownum~" inputmask="0x002000" ref="touchpad"><bounds x="13" width="1" y="~rownum~" height="1" /></element>
+			<element inputtag="TP.~rownum~" inputmask="0x004000" ref="touchpad"><bounds x="14" width="1" y="~rownum~" height="1" /></element>
+			<element inputtag="TP.~rownum~" inputmask="0x008000" ref="touchpad"><bounds x="15" width="1" y="~rownum~" height="1" /></element>
+			<element inputtag="TP.~rownum~" inputmask="0x010000" ref="touchpad"><bounds x="16" width="1" y="~rownum~" height="1" /></element>
+			<element inputtag="TP.~rownum~" inputmask="0x020000" ref="touchpad"><bounds x="17" width="1" y="~rownum~" height="1" /></element>
+			<element inputtag="TP.~rownum~" inputmask="0x040000" ref="touchpad"><bounds x="18" width="1" y="~rownum~" height="1" /></element>
+			<element inputtag="TP.~rownum~" inputmask="0x080000" ref="touchpad"><bounds x="19" width="1" y="~rownum~" height="1" /></element>
+			<element inputtag="TP.~rownum~" inputmask="0x100000" ref="touchpad"><bounds x="20" width="1" y="~rownum~" height="1" /></element>
+			<element inputtag="TP.~rownum~" inputmask="0x200000" ref="touchpad"><bounds x="21" width="1" y="~rownum~" height="1" /></element>
+			<element inputtag="TP.~rownum~" inputmask="0x400000" ref="touchpad"><bounds x="22" width="1" y="~rownum~" height="1" /></element>
+			<element inputtag="TP.~rownum~" inputmask="0x800000" ref="touchpad"><bounds x="23" width="1" y="~rownum~" height="1" /></element>
```

Comment:
You could do this with a nested `<repeat>` – remember generator parameters support `lshift` for generating the masks (e.g. see whousetc.lay for an example).